### ISSUE TITLE
About the three dataset folders

### ISSUE THREAD

**Issue by iMayuqi** [lorebianchi98/FG-OVD#2]
Hello author，
Thank you very much for the dataset you provided, which will be of great help to my future research work. Here I have two questions to ask you，hope to get your reply：
What is the relationship between benchmarks and training_sets and validation_sets? 
On which dataset is the mAP values of the model listed in the paper tested?

**Comment by lorebianchi98** [comment]
Hi! Thank you for appreciating our work!
Benchmarks, training sets, and validation sets are generated in the same way following the procedure described in the paper. The differences are the following:
- The benchmarks are generated from a subset taken from paco_lvis_v1_test.json, while training and validation sets are generated from the whole annotations, respectively, from paco_lvis_v1_val.json and paco_lvis_v1_train.json.
- The training and validation sets have only one object per vocabulary since, in the PACO dataset, it is the only way to ensure that the object's attribute does not describe other ones in the image.
- The benchmarks are manually revised.

Regarding the mAP values that you see in the paper, these values are obtained by evaluating the listed models in our benchmarks (the only available sets at the time that the paper was submitted), following these correspondances:
Difficulty-based benchmarks:
- 1_attributes.json -> Hard
- 2_attributes.json -> Medium
- 3_attributes.json -> Easy
- shuffle_negatives.json -> Trivial
Attribute-based benchmarks:
- color.json -> Color
- material.json -> Material
- pattern.json -> Pattern
- transparency.json -> Transparency

**Comment by iMayuqi** [comment]
> * >

Thank you for your reply.
In addition, I found a problem when I evaluated the groundingdino model,
args.n_hardnegatives represents the number of negative samples, so counting positive samples, all test data should be args.n_hardnegatives+1, but there is no +1 operation in groundingdino's inference code. In this way, the negative sample tested can only be args.n_hardnegatives-1

`#groundingdino code`
`vocabulary, vocabulary_id = create_vocabulary(ann, data['categories'])`
`        if args.n_hardnegatives < 10:`
`            if len(vocabulary) < args.n_hardnegatives:`
`                continue`
`            vocabulary = vocabulary[:args.n_hardnegatives]`
`            vocabulary_id = vocabulary_id[:args.n_hardnegatives]`
But I see that other model test code, such as the one below, all have +1 operations, so I guess it's a groundingdino test code error.
`#others code`
`vocabulary, vocabulary_id = create_vocabulary(ann, data['categories'])`
`        len_vocabulary = args.n_hardnegatives + 1`
`        if len(vocabulary) < len_vocabulary:`
`            continue`
`        vocabulary = vocabulary[:len_vocabulary]`
`        vocabulary_id = vocabulary_id[:len_vocabulary]`

**Comment by lorebianchi98** [comment]
Thank you for bringing this issue to my attention! You're correct; there is indeed a mistake that would affect the vocabulary length, and I will fix it :) . However, I'd like to remark that this bug doesn't impact the evaluation outcomes presented in the paper. In our experiments, we only used the GroundingDINO model with the parameter n_hardnegatives set to 10. As outlined in the paper, GroundingDINO follows a unique evaluation method. Unlike other detectors, where we conduct inference for each vocabulary in the benchmarks, GroundingDINO takes a different approach. We perform inference for every caption within each vocabulary. This strategy enables us to run the inference code with the n_hardnegatives = 10 argument only once per benchmark (thus, by circumventing the malfunctioning 'if' condition). By storing all predictions for each vocabulary length, we reduced the prediction set only during evaluation, thereby reducing inference time.

Anyway, I will fix this error!
